xarray/core/datatree.py

```diff
@@ -1486,6 +1490,50 @@ def groups(self):
         """Return all groups in the tree, given as a tuple of path-like strings."""
         return tuple(node.path for node in self.subtree)
 
+    def _unary_op(self, f, *args, **kwargs) -> DataTree:
+        # TODO do we need to any additional work to avoid duplication etc.? (Similar to aggregations)
+        return self.map_over_subtree(f, *args, **kwargs)  # type: ignore[return-value]
+
+    def _binary_op(self, other, f, reflexive=False, join=None) -> DataTree:
+        from xarray.core.dataset import Dataset
+        from xarray.core.groupby import GroupBy
+
+        if isinstance(other, GroupBy):
+            # TODO should we be trying to make this work?
+            raise NotImplementedError
```

Comment:
```suggestion
        return NotImplemented
```

`NotImplemented` is a sentinel value that tells Python that an arithmetic operator is not implemented, and allows the other argument to try implementing it. If all special methods return `NotImplemented`, then Python raises an informative `TypeError`.

Should we also explicitly exclude `Dataset` here, or are the "mapped over all nodes" semantics of DataTree + Dataset arithmetic obvious enough? https://github.com/pydata/xarray/issues/9365

Reply:
> If all special methods return NotImplemented, then Python raises an informative TypeError.

Ah I did not realise that part. So `NotImplemented` is fine here for now.

> Should we also explicitly exclude Dataset here, or are the "mapped over all nodes" semantics of DataTree + Dataset arithmetic obvious enough?

I think it's fine to allow `Dataset` - I did that deliberately. It's also tested now.

Reply:
done in https://github.com/pydata/xarray/pull/9619/commits/9e2dfade71d1d1eb22969023b5924834c104db92